src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1172,18 +1528,38 @@ def find_existing_acr(cmd, app: "ContainerApp"):
     return None, None
 
 
-def check_env_name_on_rg(cmd, managed_env, resource_group_name, location):
+def check_env_name_on_rg(cmd, env, resource_group_name, location, custom_location_id=None, connected_cluster_id=None):
+    env_name = env
+    resource_type = None
+    if is_valid_resource_id(env):
+        env_dict = parse_resource_id(env)
+        env_name = env_dict.get("name")
+        resource_group_name = env_dict.get("resource_group")
+        resource_type = env_dict.get("resource_type")
+    if resource_type is None:
+        if custom_location_id or connected_cluster_id:
+            resource_type = CONNECTED_ENVIRONMENT_TYPE
     if location:
         _ensure_location_allowed(cmd, location, CONTAINER_APPS_RP, "managedEnvironments")
-    if managed_env and resource_group_name and location:
+    if env and resource_group_name:
         env_def = None
         try:
-            env_def = ManagedEnvironmentClient.show(cmd, resource_group_name, parse_resource_id(managed_env)["name"])
-        except:  # pylint: disable=bare-except
-            pass
+            if resource_type and CONNECTED_ENVIRONMENT_TYPE.lower() == resource_type.lower():
+                env_def = ConnectedEnvironmentClient.show(cmd, resource_group_name, env_name)
+            else:
+                env_def = ManagedEnvironmentPreviewClient.show(cmd, resource_group_name, env_name)
+        except Exception as e:  # pylint: disable=bare-except
+            handle_non_404_status_code_exception(e)
         if env_def:
-            if format_location(location) != format_location(env_def["location"]):
-                raise ValidationError("Environment {} already exists in resource group {} on location {}, cannot change location of existing environment to {}.".format(parse_resource_id(managed_env)["name"], resource_group_name, env_def["location"], location))
+            if location and location != format_location(env_def["location"]):
```

Comment:
why not format the location?

Reply:
Yes, need to format the location, will fix it.

Reply:
Done.